src/main/cpp/src/zorder.cu

```diff
@@ -114,4 +226,53 @@ std::unique_ptr<cudf::column> interleave_bits(
     mr);
 }
 
+std::unique_ptr<cudf::column> hilbert_index(
+  int32_t const num_bits_per_entry,
+  cudf::table_view const& tbl,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr) {
+ 
+  auto num_rows = tbl.num_rows();
+  auto num_columns = tbl.num_columns();
+
+  CUDF_EXPECTS(num_bits_per_entry > 0 && num_bits_per_entry <= 32, "the number of bits must be >0 and <= 32.");
+  CUDF_EXPECTS(num_bits_per_entry * num_columns <= 64, "we only support up to 64 bits of output right now.");
+  CUDF_EXPECTS(num_columns > 0, "at least one column is required.");
+
+  CUDF_EXPECTS(
+    std::all_of(tbl.begin(),
+                tbl.end(),
+                [](cudf::column_view const& col) { return col.type().id() == cudf::type_id::INT32; }),
+    "All columns of the input table must be INT32.");
+
+  auto input_dv = cudf::table_device_view::create(tbl, stream);
+
+  auto output_data_col = cudf::make_numeric_column(
+      cudf::data_type{cudf::type_id::INT64}, num_rows, cudf::mask_state::UNALLOCATED, stream, mr);
+
+  auto output_dv_ptr = cudf::mutable_column_device_view::create(*output_data_col, stream);
+
+  thrust::for_each_n(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator<cudf::size_type>(0),
+    num_rows,
+    [output_col = *output_dv_ptr,
+     num_bits_per_entry,
+     num_columns,
+     input = *input_dv] __device__ (cudf::size_type row_index) {
+       long_backed_array row(num_bits_per_entry);
```

Comment:
`const`.

Reply:
We modify the row. Why would I mark it as `const` to say that we cannot modify it?